### PR TITLE
migrate services from `http::Request` to `azure_core::Request`

### PR DESCRIPTION
**Pull request by ctaggart**
This is a subset of #833 to make both easier to review. This migrates just the services.

The check of all features passed: https://github.com/ctaggart/azure-sdk-for-rust/actions/runs/2525015263

### CODE REVIEW

**Review by rylev — APPROVED** [review]

File: sdk/core/src/request.rs [hunk 1]
```diff
@@ -94,6 +94,30 @@ impl Request {
     pub fn set_body(&mut self, body: impl Into<Body>) {
         self.body = body.into();
     }
+
+    pub fn insert_header<K, V>(&mut self, key: K, value: V)
```
[the remark this answers is not in the document]
Comment (ctaggart) [reply sdk/core/src/request.rs:98]: Looking at [http_types::Request](https://docs.rs/http-types/latest/http_types/struct.Request.html), it has both.

File: services/autorust/codegen/src/codegen_operations.rs [hunk 2]
```diff
@@ -362,13 +364,13 @@ impl ToTokens for BuildRequestParamsCode {
                                 if param.is_string(){
                                     quote! {
                                         for value in &this.#param_name_var {
-                                            url.query_pairs_mut().append_pair(#param_name, value);
+                                            req.url_mut().query_pairs_mut().append_pair(#param_name, value);
```
[the remark this answers is not in the document]
Comment (ctaggart) [reply services/autorust/codegen/src/codegen_operations.rs:367]: I think that would be helpful.

File: sdk/core/src/request.rs [hunk 3]
```diff
@@ -119,3 +119,17 @@ impl Request {
         self.insert_header(item.name(), item.value())
     }
 }
+
+/// Temporary hack to convert preexisting requests into the new format. It
+/// will be removed as soon as we remove the dependency from `http::Request`.
```
[the remark this answers is not in the document]
Comment (ctaggart) [reply sdk/core/src/request.rs:124]: I'm removing it in #833. I just had to add it back here to make this subset compile.

File: sdk/core/src/request.rs [hunk 4]
```diff
@@ -94,6 +94,30 @@ impl Request {
...
+
+    pub fn insert_header<K, V>(&mut self, key: K, value: V)
+    where
+        K: Into<crate::headers::HeaderName>,
+        V: Into<crate::headers::HeaderValue>,
+    {
+        self.headers.insert(key, value)
+    }
+
+    pub fn add_optional_header_ref<T: crate::Header>(&mut self, item: &Option<&T>) {
```
[the remark this answers is not in the document]
Comment (cataggar) [reply sdk/core/src/request.rs:106]: I have this change ready for a follow-up PR after #833.

**Pull request merged**